src/table/styling-utils.js

```diff
@@ -15,14 +15,12 @@ export function getHeadStyle(layout, theme) {
 
 export function getBodyStyle(layout, theme) {
   const content = layout.components?.[0]?.content;
-  const hoverColor = layout.components?.[0]?.hoverColor;
-  const hoverFontColor = layout.components?.[0]?.hoverFontColor;
   return content
     ? {
         fontColor: getColor(content.fontColor, '#404040', theme),
         fontSize: content.fontSize || '14px',
-        hoverBackGroundColor: `${getColor(hoverColor, 'blue', theme)} !important`,
-        hoverFontColor: getColor(hoverFontColor, 'black', theme),
+        hoverBackGroundColor: `${getColor(content.hoverColor, '#D3D3D3', theme)} !important`,
+        hoverFontColor: getColor(content.hoverFontColor, 'black', theme),
```

Comment:
This still overrides `fontColor`, so `this.fontColor` (not sure you can use `this` here, but something similar) should be used as default value

Reply:
What do you mean? If I understand correctly, the hoverFontColor should override fontColor in tableCell, if the hover is taking effect.

Reply:
Only if there is a `hoverFontColor` set. We want the same behaviour as the old table. I just looked at it and it is a bit more complex than I thought:
- If no font color is set (hover or non-hover) and a hover background color is set -> the font color becomes white when the background color is a darker shade. There is logic for this inside the sn-action-button we could reuse.
- If you set a hover font color but no hover background color -> only a font color is applied when hovering
- If hovering is turned on but no hover colors are set -> use the defualt light gray background color and don't change the font color

I think that was all. I'd say have a look in sense in the old table that you get all the behaviours. This should have been specified in the issue, sorry about finding out about this now.

Reply:
no worries, good points really. Not thinking of all these before. Will look into all those cases.